### ISSUE TITLE
creditmemo_created event registered twice if creditmemo mail is sent async

### ISSUE THREAD

**Issue by Jeroenll** [magento/inventory#3304]
### Preconditions (*)
1. Setup order/creditmemo mail sending in Magento.
2. Enable async mail sending.

### Steps to reproduce (*)
1. Place and invoice order.
2. Create creditmemo based on invoice, return items to stock and send copy of creditmemo as email to customer.
3. Order is now closed and `creditmemo_created` event is saved in inventory_reservation table.
4. Run `sales_send_order_creditmemo_emails` cronjob
5. Another `creditmemo_created` event is saved in inventory_reservation table.

### Expected result (*)
1. `creditmemo_created` event is saved only once.

### Actual result (*)
1. `creditmemo_created` event is saved twice.

**Comment by Jeroenll** [comment]
It's fixed with this commit: https://github.com/magento/magento2/commit/2f8b94d171dbb8c0847ffe7e6b3b013431c233c3 but the issue remains that on every creditmemo save new `creditmemo_created` entries are added.

Should we change this event (https://github.com/magento/inventory/blob/1.2-develop/InventorySales/etc/events.xml#L15) to `sales_order_creditmemo_refund`?

**Comment by erfanimani** [comment]
I agree that the event should be changed from `sales_order_creditmemo_save_after` to something else. Or at the very least have a check in that event to ensure there is no reservation compensation tied to this particular SKU/credit memo combination. However, the latter seems more complex/over-engineered than simply changing it to an event that only fires once per credit memo.

That said, the event observer can also be wrapped into an if statement with condition `$creditmemo->isObjectNew();`, see https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Model/AbstractModel.php#L680

Creating a new reservation compensation every time the object is saved without any idempotency check is just waiting for obscure issues to happen..

Relevant https://github.com/magento/magento2/pull/31189

**Comment by PascalBrouwers** [comment]
The problem is not only with creditmemo's but also order reservations, shipments, etc. Some reservations will be double because it does not check if such a reservation already exists.

**Comment by ioweb-gr** [comment]
I noticed that on the 
https://github.com/magento/inventory/blob/develop/InventoryShipping/Observer/SourceDeductionProcessor.php

There's a condition that checks if the shipment is new 

https://github.com/magento/inventory/blob/0f0eaf1e19dbea1a660b8de66eb8873a4c71ccab/InventoryShipping/Observer/SourceDeductionProcessor.php#L96

Unlike the 

https://github.com/magento/inventory/blob/develop/InventoryShipping/Observer/VirtualSourceDeductionProcessor.php

In my case I'm facing this only on the virtual products, not on physical ones.
